runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -299,6 +299,66 @@ public Response submitEvent(@PathParam("machineId") String machineId,
         return Response.status(Response.Status.ACCEPTED).build();
     }
 
+    /**
+     * Update EventData of the specified Event name under the specified State machine
+     *
+     * @param machineId the state machine identifier
+     * @return Response with execution status code
+     * @throws Exception
+     */
+    @POST
+    @Path("/{machineId}/context/eventupdate")
+    @Timed
+    public Response updateEvent(@PathParam("machineId") String machineId,
+                                EventData eventData
+    ) throws Exception {
+        try {
+            LoggingUtils.registerStateMachineIdForLogging(machineId);
+            if (eventData.getData() == null || eventData.getName() == null || eventData.getEventSource() == null) {
+                return Response.status(Response.Status.BAD_REQUEST.getStatusCode()).entity(
+                        "Event Data|Name|Source cannot be null.").build();
+            }
+            if (eventsDAO.findBySMIdAndName(machineId, eventData.getName()).getStatus() != Event.EventStatus.triggered) {
+                return Response.status(Response.Status.FORBIDDEN.getStatusCode()).entity(
+                        "Input event is not in triggered state.").build();
+            }
+            boolean canUpdateEventData = false;
+            /* List holds objects retrieved from States matching input machineId and eventName as [taskId, machineId, status] */
+            List<Object[]> states = statesDAO.findStatesByDependentEvent(machineId, eventData.getName());
+
+            for (Object[] state : states) {
+                if (state[2] == Status.running) {
```

Comment:
switch case as mentioned earlier by Kamesh's review.

Reply:
As per design, only triggered events with at least one of the next/immediate dependent state(s) in errored/sidelined/initialized state and none of these states in running state can be updated. 
If we use switch case, status can be categorised in 3:
1. Running - Update not allowed irrespective of what is the status of other states.
2. [ Completed, Cancelled, Unsidelined ] - Update not allowed for this state but event data can be updated if at least one of the other dependent states is [ sidelined, errored, initialized ] and none of them is in [running] status.
3. [ sidelined, errored, initialized ] - Event Data update allowed if none of the dependent states is in [running] status.

Since, states can be iterated in the loop/switch case in any order, maintaining results with only one flag 'canUpdateEventData' doesn't help. Also adding flag for encountered 'running' state doesn't help.

Consider one failure scenario, Say dependent states are as: 
[ T1, initialized], [T2, completed] - Event data can be updated.
    Loop iteration order:
    1. [T1, T2] - Will not update (cannot be handled in switch unless we make it more complex by adding few more flags).
    2. [T2, T1] - Will update.

For switch case defined as  :
for (Object[] state : states) {
                Status status = (Status)state[2];
                switch (status) {
                    case running:
                        encounteredRunningState = true;
                        canUpdateEventData = false;
                        break;
                    case initialized:
                    case errored:
                    case sidelined:
                        canUpdateEventData = true;
                        break;
                    case completed:
                    case cancelled:
                        canUpdateEventData = false;
                        break;
                }
                if(encounteredRunningState) {
                    break;
                }
            }

However, in existing if-else, it's handled if 'running' status encountered, mark flag to false and break (update invalid) else 'initilized|sidelined|errored', mark flag to true but don't break for loop because we may encounter 'running' state later.